src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -0,0 +1,206 @@
+                return wtPath;
+            }
+        }
+        CATCH_LOG();
+
+        // If we're here, then we couldn't resolve our exe from the package. This
+        // means we're running unpackaged. We should just use the
+        // WindowsTerminal.exe that's sitting in the directory next to us.
+        try
+        {
+            HMODULE hModule = GetModuleHandle(nullptr);
+            THROW_LAST_ERROR_IF(hModule == nullptr);
+            std::wstring dllPathString;
+            THROW_IF_FAILED(wil::GetModuleFileNameW(hModule, dllPathString));
+            const std::filesystem::path dllPath{ dllPathString };
+            const std::filesystem::path rootDir = dllPath.parent_path();
+            std::filesystem::path wtPath = rootDir / "WindowsTerminal.exe";
+            return wtPath;
+        }
+        CATCH_LOG();
+
+        return L"wt.exe";
+    }();
+    return exePath;
+}
+
+// Method Description:
+// - This method is called when the user activates the context menu item. We'll
+//   launch the Terminal using the current working directory.
+// Arguments:
+// - psiItemArray: a IShellItemArray which contains the item that's selected.
+// Return Value:
+// - S_OK if we successfully attempted to launch the Terminal, otherwise a
+//   failure from an earlier HRESULT.
+HRESULT OpenTerminalHere::Invoke(IShellItemArray* psiItemArray,
+                                 IBindCtx* /*pBindContext*/)
+{
+    DWORD count;
+    psiItemArray->GetCount(&count);
+
+    winrt::com_ptr<IShellItem> psi;
+    RETURN_IF_FAILED(psiItemArray->GetItemAt(0, psi.put()));
+
+    PWSTR pszName;
+    RETURN_IF_FAILED(psi->GetDisplayName(SIGDN_FILESYSPATH, &pszName));
+
+    {
+        wil::unique_process_information _piClient;
+        STARTUPINFOEX siEx{ 0 };
+        siEx.StartupInfo.cb = sizeof(STARTUPINFOEX);
+
+        std::wstring cmdline = fmt::format(L"\"{}\" -d \"{}\"", _getExePath(), pszName);
+        RETURN_IF_WIN32_BOOL_FALSE(CreateProcessW(
+            nullptr,
+            cmdline.data(),
+            nullptr, // lpProcessAttributes
+            nullptr, // lpThreadAttributes
+            false, // bInheritHandles
+            EXTENDED_STARTUPINFO_PRESENT | CREATE_UNICODE_ENVIRONMENT, // dwCreationFlags
+            nullptr, // lpEnvironment
+            nullptr,
+            &siEx.StartupInfo, // lpStartupInfo
+            &_piClient // lpProcessInformation
+            ));
+    }
+
+    CoTaskMemFree(pszName);
+
+    return S_OK;
+}
+
+HRESULT OpenTerminalHere::GetToolTip(IShellItemArray* /*psiItemArray*/,
+                                     LPWSTR* ppszInfoTip)
+{
+    // tooltip provided here, in this case none is provided
+    *ppszInfoTip = nullptr;
+    return E_NOTIMPL;
+}
+
+HRESULT OpenTerminalHere::GetTitle(IShellItemArray* /*psiItemArray*/,
+                                   LPWSTR* ppszName)
+{
+    // Change the string we return depending on if we're running from the dev
+    // build package or not.

```

Comment:
nit: move this up to be a `// Method Description`. Or not, I don't really care... Most of the functions here do that (presumably because of the sample).

Reply:
meh, I didn't want to give the impression I did anything intentional with those. Rather just make them doc comments when there's something to document 😉